js/source/tile.js

```diff
@@ -32,12 +33,11 @@ Tile.prototype = {
      * @returns {Object} position
      * @private
      */
-    positionAt: function(coord, sourceMaxZoom) {
-        coord = coord.zoomTo(Math.min(this.coord.z, sourceMaxZoom));
+    positionAt: function(coord) {
+        var zoomedCoord = coord.zoomTo(Math.min(this.coord.z, this.sourceMaxZoom));
         return {
-            x: (coord.column - this.coord.x) * this.tileExtent,
-            y: (coord.row - this.coord.y) * this.tileExtent,
-            scale: this.scale
```

Comment:
I think it makes more sense for `TilePyramid` to calculate `scale` because the `TilePyramid` is necessarily coupled to the `Transformation` while `Tile` is not.